modules/codegen/src/smithy4s/codegen/Renderer.scala

```diff
@@ -415,6 +433,66 @@ private[codegen] class Renderer(compilationUnit: CompilationUnit) { self =>
     ).addImports(imports)
   }
 
+  private def renderProductMixin(
+      name: String,
+      originalName: String,
+      fields: List[Field],
+      mixins: List[Type],
+      recursive: Boolean,
+      hints: List[Hint],
+      mixinExtensions: Line,
+      adtParent: Option[String],
+      additionalLines: Lines
+  ): Lines = {
+    block(line"trait $name$mixinExtensions") {
+      lines(
+        // TODO: noDefault is probably not the right move, just putting for quick sanity check
+        fields.map(f => line"def ${fieldToRenderLine(f, noDefault = true)}")
+      )
+    }
+  }
+
+  private def renderProduct(
```

Comment:
let's just pass the whole `Product` instance as a parameter to these methods, they're starting to be too wordy